lib/stores/filesystem.js

```diff
@@ -48,6 +48,19 @@ class FilesystemStore {
     return path.join(this.rootDirectory, bucketName);
   }
 
+  async makeBucketDirIfNotExists(bucketPath) {
+    let exists = false;
+    try {
+      await fs.access(bucketPath, fs.constants.F_OK);
+    } catch (ex) {
+      exists = true;
```

Comment:
Thanks for refactoring this, I wasn't sure exactly where the Windows tests were going wrong since I replaced fs-extra.

Could you move this to the utils file as `ensureDir(path)`? You should also check that the exception code is `ENOENT` and rethrow when it's not. You can also move the `mkdir` call below directly to this catch block instead of storing a temporary variable.